### PR TITLE
Fix the dry-run flag behaviour

### PR DESCRIPTION
**Pull request by johnbillion**
See #4.

Still needs more work.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -80,14 +80,14 @@ runs:
       shell: bash
 
     - name: Generate attestation for the plugin
-      if: ${{ ! inputs.dry-run }}
+      if: ${{ inputs.dry-run == false }}
```
Comment (szepeviktor) [inline action.yml]: I would write `!= true`

**Comment by szepeviktor** [comment]
Here we go
https://github.com/johnbillion/user-switching/actions/runs/11891386423/job/33132043839
✈️

**Comment by szepeviktor** [comment]
You can always add a temporary debug step
```yaml
run: echo dry-run=${{ toJSON(inputs.dry-run) }}
# or
run: echo inputs=${{ toJSON(inputs) }}
```

**Comment by johnbillion** [comment]
@szepeviktor What am I doing wrong? This run has the `dry-run` input set to true but the steps with the `if` conditions still ran: https://github.com/johnbillion/user-switching/actions/runs/11891945602/job/33133767062 .

The input values can be seen at the top by expanding the arrow next to `Run johnbillion/action-wordpress-plugin-attestation@dry-run`:

```
Run johnbillion/action-wordpress-plugin-attestation@dry-run
  with:
    version: 1.8.0
    zip-path: /home/runner/work/user-switching/user-switching/user-switching.zip
    dry-run: true
    plugin: user-switching
    timeout: 60
```

**Comment by szepeviktor** [comment]
Try that debug step!

**Comment by johnbillion** [comment]
Oh the value is being set to string `"true"` instead of a boolean.

```
"Inputs: {
    "version": "1.8.0",
    "zip-path": "/home/runner/work/user-switching/user-switching/user-switching.zip",
    "dry-run": "true",
    "plugin": "user-switching",
    "timeout": "60"
}"
```

**Comment by szepeviktor** [comment]
https://github.com/actions/runner/issues/1483#issuecomment-994747723 👀

**Comment by johnbillion** [comment]
Yep that looks correct. Inputs for actions are strings only: https://stackoverflow.com/a/76294014

**Comment by johnbillion** [comment]
Okay we're looking good now. What a pickle.

* ✅ `dry-run` true: https://github.com/johnbillion/user-switching/actions/runs/11892353362/job/33135038954
* ✅ `dry-run` false: https://github.com/johnbillion/user-switching/actions/runs/11892395968/job/33135165939

**Comment by szepeviktor** [comment]
Actually `echo "` hides quotation marks in the JSON.
```
printf 'Inputs = '
cat <<"GITHUB__CONTEXT"
${{ toJSON(inputs) }}
GITHUB__CONTEXT
```

[_source_](https://github.com/szepeviktor/byte-level-care/blob/494b55db242ff32e7c61f5d3a86404256cb12781/.github/workflows/workflow.yml#L48-L54)

**Comment by johnbillion** [comment]
It shows up correctly when you expand the `>` next to the command, but yeah the output is not clear.

**Comment by johnbillion** [comment]
Single quotes save the day.

**Comment by johnbillion** [comment]
Thanks for the assistance @szepeviktor 💪

**Pull request merged**